src/sentry/api/endpoints/project_dynamic_sampling.py

```diff
@@ -243,60 +301,127 @@ def get(self, request: Request, project) -> Response:
             # If the number of the projects in the breakdown is greater than 10 projects,
             # then a question needs to be raised on the eligibility of the org for dynamic sampling
-            if len(project_breakdown) > 10:
+            if len(projects_counts) > 10:
                 return Response(
                     status=status.HTTP_400_BAD_REQUEST,
                     data={
                         "detail": "Way too many projects in the distributed trace's project breakdown"
                     },
                 )
 
+            # Calculating the count of projects that started a distributed trace
+            # in the fetched randomly sampled transactions
+            root_projects_count = len(
+                [project for project in projects_counts if project["root_count"] > 0]
+            )
+            # get requested project id in projects_count dataset
+            requested_project = next(
+                item for item in projects_counts if item["project_id"] == project.id
+            )
+
+            if root_projects_count >= 1:
+                parent_project_breakdown = []
+                total_count = sum(
+                    _project["count"] for _project in projects_counts if _project["root_count"] > 0
+                )
+                for _project in projects_counts:
+                    if _project["root_count"] > 0:
+                        parent_project_breakdown.append(
+                            {
+                                "project": _project["project"],
+                                "project_id": _project["project_id"],
+                                "percentage": _project["root_count"] / total_count,
+                            }
+                        )
+
+            else:
+                parent_project_breakdown = []
+
+            # Get project breakdown
+            project_breakdown = []
+
+            parent_project_ids = {p["project_id"] for p in parent_project_breakdown}
+
+            if requested_project["project_id"] in parent_project_ids:
+                if len(parent_project_breakdown) > 1:
+
+                    # loop over trace ids, and filter down trace ids where project is sentry
+                    parent_trace_ids = [
+                        transaction.get("trace")
+                        for transaction in transactions
+                        if transaction.get("is_root", False)
+                        and transaction["project_id"] == project.id
+                    ]
+                    # Calculate parent project_breakdown attribute:
+                    project_breakdown = self.__run_discover_query(
+                        columns=[
+                            "project_id",
+                            "project",
+                            "count()",
+                        ],
+                        query=f"event.type:transaction !has:trace.parent_span trace:[{','.join(parent_trace_ids)}]",
```

Comment:
Why do you need this condition if you already have `parent_trace_ids`?
```suggestion
                        query=f"event.type:transaction trace:[{','.join(parent_trace_ids)}]",
```